### PR TITLE
Add description to interpreter.Value

### PR DESCRIPTION
**Pull request by turbolent**
______

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [x] Code follows the [standards mentioned here](https://github.com/onflow/cadence/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels

### CODE REVIEW

**Review by dreamsmasher — APPROVED** [review]
Ahh, thank you for adding these to the docs in response to my question earlier!

**Comment by bluesign** [comment]
What is the suggested way to convert from interpreter.Value to cadence.Value?

**Comment by turbolent** [comment]
@bluesign Good question, adding it to the comment.

File: runtime/interpreter/value.go [hunk 1]
```diff
@@ -89,8 +89,16 @@ func (NonStorable) ChildStorables() []atree.Storable {
...
+// Value is the Cadence value hierarchy which is heavily tied to the interpreter and persistent storage,
+// and has lots of implementation details.
+//
+// We do not want to expose those details to users
+// (for example, Cadence is used as a library in flow-go (FVM), in the Flow Go SDK, etc.),
+// because we want to be able to change the API and implementation details;
+// nor do we want to require users to Cadence (the library) to write lots of low-level/boilerplate code (e.g. setting up storage).
+//
+// To accomplish this, cadence.Value is the "user-facing" hierarchy that is easy to work with:
+// simple Go types that can be used without an interpreter or storage.
```
Comment (turbolent) [inline runtime/interpreter/value.go:101]:
```suggestion
// simple Go types that can be used without an interpreter or storage.
//
// cadence.Value can be converted to an interpreter.Value by "importing" it with importValue,
// and interpreter.Value can be "exported" to a cadence.Value with ExportValue.
```

**Comment by bluesign** [comment]
@turbolent thanks a lot, it is maybe unrelated or I maybe missing some context, but can we say that eventually everything will be importable and exportable? 

I stuck once on this, though been very long time I am fuzzy on details, some values were not importable etc

**Review by dsainati1 — APPROVED** [review]

**Comment by turbolent** [comment]
@bluesign it should be possible to make more value importable or exportable that are currently not yet. 

We flagged some types as not importable/exportable intentionally, just because that requires extra effort, but adding support is a low priority. For some values we decided that we do not want to make them importable/exportable yet because we still want to change details about them.

For some values it is easier to add support for importing/exporting, as their internal representation is simpler, for some it is harder. For others, like functions, it is unlikely that they will ever gain support for importing/exporting, because it is unclear how this could be implemented (how are functions serialized? as source code? what about closures?)

**Pull request merged**